### PR TITLE
[MINOR][Core] Fix nits in MetricsSystemSuite

### PR DESCRIPTION
**Pull request by vundela**
## What changes were proposed in this pull request?
Fixing nits in MetricsSystemSuite file
1) Using Sink instead of Source while casting 
2) Using meaningful naming for variables, which reflect their usage

## How was this patch tested?
Ran the tests locally and all of them passing

### CODE REVIEW

**Comment by vanzin** [comment]
Please:
* remove all of the template from the PR summary
* explain which nits are being fixed

**Comment by vanzin** [comment]
ok to test

File: core/src/test/scala/org/apache/spark/metrics/MetricsSystemSuite.scala [hunk 1]
```diff
@@ -42,7 +43,7 @@ class MetricsSystemSuite extends SparkFunSuite with BeforeAndAfter with PrivateM
     val metricsSystem = MetricsSystem.createMetricsSystem("default", conf, securityMgr)
     metricsSystem.start()
     val sources = PrivateMethod[ArrayBuffer[Source]]('sources)
-    val sinks = PrivateMethod[ArrayBuffer[Source]]('sinks)
+    val sinks = PrivateMethod[ArrayBuffer[Sink]]('sinks)
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/metrics/MetricsSystemSuite.scala]: Is this actually a bug fix?

Comment (vundela) [reply core/src/test/scala/org/apache/spark/metrics/MetricsSystemSuite.scala]: I thought the same, but tests are passing. So decided as nit fixes.

**Comment by vundela** [comment]
Thanks for review @vanzin, @srowen.  Added the appropriate summary.

**Review by jiangxb1987 — APPROVED** [review]
LGTM

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**